pkg/k8shandler/statefulset.go

```diff
@@ -242,15 +248,15 @@ func (node *statefulSetNode) restart(upgradeStatus *api.ElasticsearchNodeStatus)
 
 	if upgradeStatus.UpgradeStatus.UpgradePhase == api.NodeRestarting {
 
-		err, ordinal := node.partition()
+		ordinal, err := node.partition()
 		if err != nil {
 			logrus.Infof("Unable to get node ordinal value: %v", err)
 			return
 		}
 
 		for index := ordinal; index > 0; index-- {
 			// get podName based on ordinal index and node.name()
-			podName := fmt.Sprintf("%v-%v", node.name(), index)
+			podName := fmt.Sprintf("%v-%v", node.name(), index-1)
```

Comment:
are you sure we should be doing `-1`? i thought the ordinal index is the same as the index we see in the pod name..
example: https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#rolling-out-a-canary